ui/app/abilities/variable.js

```diff
@@ -116,22 +116,29 @@ export default class Variable extends AbstractAbility {
 
   @computed('allPaths', 'namespace', 'path', 'token.selfTokenPolicies')
   get policiesSupportVariableWriting() {
-    if (this.namespace === WILDCARD_GLOB && this.path === WILDCARD_GLOB) {
-      // If you're checking if you can write from root, and you don't specify a namespace,
-      // Then if you can write in ANY path in ANY namespace, you can get to /new.
+    if (this.path === WILDCARD_GLOB) {
+      // If checking for write permission on the root path
       return this.policyNamespacesIncludeVariablesCapabilities(
         this.token.selfTokenPolicies,
         ['write'],
-        this._nearestMatchingPath(this.path)
+        WILDCARD_GLOB
       );
```

Comment:
It's not clear to me why we need to special case the root path? Does `_nearestMatchingPath` and the ACL checks not handle this themselves? If not, should we push this logic down into those?

Reply:
This ended up being a clue for me that I was barking up the wrong tree here.

I've since made the fix for the customer issue in the template file (instead of the "can I write?" check on the index page being namespace-dropdown dependent, it's a `*`, so if they can write in any namespace they get a "Create New Variable" button)

Then, changed the `can${write|read|destroy}` checks to be bimodal:
- if namespace is `*`, use `policyNamespacesIncludeVariablesCapabilities()` to see if any of the policies' namespaces' rules include the verb at the provided path
- if namespace is specified, run a filter on `allVariablePathRules[]` to check for namespace, path, and capability.